databroker/experimental/server_ext.py

```diff
@@ -0,0 +1,501 @@
+        # TODO Apply pagination, as we do in Databroker.
+        for doc in list(
+            self.collection.find(
+                # self._build_mongo_query({"active": True}), {"uid": True}
+                self._build_mongo_query({"data_url": {"$ne": None}}),
+                {"uid": True},
+            )
+        ):
+            yield doc["uid"]
+
+    def __len__(self):
+        return self.collection.count_documents(
+            # self._build_mongo_query({"active": True})
+            self._build_mongo_query({"data_url": {"$ne": None}})
+        )
+
+    def __length_hint__(self):
+        # https://www.python.org/dev/peps/pep-0424/
+        return self.collection.estimated_document_count(
+            # self._build_mongo_query({"active": True}),
+            self._build_mongo_query({"data_url": {"$ne": None}}),
+        )
+
+    def __repr__(self):
+        # Display up to the first N keys to avoid making a giant service
+        # request. Use _keys_slicer because it is unauthenticated.
+        N = 10
+        return tree_repr(self, self._keys_slice(0, N, direction=1))
+
+    def search(self, query):
+        """
+        Return a MongoAdapter with a subset of the mapping.
+        """
+        return self.query_registry(query, self)
+
+    def sort(self, sorting):
+        return self.new_variation(sorting=sorting)
+
+    def _keys_slice(self, start, stop, direction):
+        assert direction == 1, "direction=-1 should be handled by the client"
+        skip = start or 0
+        if stop is not None:
+            limit = stop - skip
+        else:
+            limit = None
+        for doc in self.collection.find(
+            # self._build_mongo_query({"active": True}),
+            self._build_mongo_query({"data_url": {"$ne": None}}),
+            skip=skip,
+            limit=limit,
+        ):
+            yield doc["uid"]
+
+    def _items_slice(self, start, stop, direction):
+        assert direction == 1, "direction=-1 should be handled by the client"
+        skip = start or 0
+        if stop is not None:
+            limit = stop - skip
+        else:
+            limit = None
+
+        for doc in self.collection.find(
+            # self._build_mongo_query({"active": True}),
+            self._build_mongo_query({"data_url": {"$ne": None}}),
+            skip=skip,
+            limit=limit,
+        ):
+            if doc["structure_family"] == StructureFamily.array:
+                yield (
+                    doc["uid"],
+                    WritingArrayAdapter(self.database, self.directory, doc),
+                )
+            elif doc["structure_family"] == StructureFamily.dataframe:
+                yield (
+                    doc["uid"],
+                    WritingDataFrameAdapter(self.database, self.directory, doc),
+                )
+            else:
+                raise ValueError("Unsupported Structure Family value in the databse")
+
+    def _item_by_index(self, index, direction):
```

Comment:
I think this can be implemented in terms of `_items_slice`

Reply:
Yeah, this occurred to me recently. I'm not sure what I was thinking here.

Reply:
is there a reason to require this to be defined separately for the IndexersMixin interface in tiled?

Reply:
I like the idea of having a method with this name. The implementation of this would make things easier for someone using this class. I think that calling `_item_slice` from within `_item_by_index` is a good approach.

Reply:
I think this needs to return a single item, not a list or iterable of items. That is, we want `thing`, not `[thing]`. I think this currently returns `[thing]`.